docs/CONTRIBUTING.md

```diff
@@ -1,62 +1,47 @@
 # Contributing guidelines
 
-## Pull Request Checklist
-
-Before sending your pull requests, make sure you followed this list.
-
-- Read [contributing guidelines](CONTRIBUTING.md).
-- Read [Code of Conduct] [TBD]
-- Check if my changes are consistent with the [guidelines]: TBD
-- Changes are consistent with the [Coding Style] (TBD)
-- Run [Unit Tests]: ``nose2``
-
-## How to become a contributor and submit your own code
-
-
-### Contributing code
-
-If you have improvements to graphengine, send us your pull requests! For those
-just getting started, Github has a
-[how to](https://help.github.com/articles/using-pull-requests/).
-
-Before sending your pull request for
-[review](https://https://github.com/adatao/graphengine/pulls),
-make sure your changes are consistent with the guidelines and follow the
-graphengine coding style (TBD).
-
 #### General guidelines and philosophy for contribution
 
-*   Include unit tests when you contribute new features, as they help to a)
-    prove that your code works correctly, and b) guard against future breaking
-    changes to lower the maintenance cost.
-*   Bug fixes also generally require unit tests, because the presence of bugs
-    usually indicates insufficient test coverage.
-*   Keep API compatibility in mind when you change code in core GraphEngine. 
-*   Tests should follow the basic principles:
-**  Only depend on what you use in your requirements. 
-**  All code should have unit tests.
-**  Each class should have its own unit test file. 
-**  Smaller tests are better. 
-**  Always use seed if you're using any stochasticity functions. 
-**  Avoid using sleep in multi thread tests.
-**  Avoid using external resource in tests e.g: S3, requests to external services: download model from repos etc. 
-
-#### License
-
-TBD
-
-#### Python coding style
-
-Changes to code should conform to
+* Include unit tests when you contribute new features, as they help to
+  - prove that your code works correctly,
+  - guard against future breaking changes
+* Bug fixes also generally require unit tests, because the presence of bugs usually indicates insufficient test
+  coverage.
+* Keep API compatibility in mind when you change code in core GraphEngine.
+* Tests should follow the basic principles:
+  -  Only depend on what you use in your requirements.
+  -  All code should have unit tests.
+  -  Each class should have its own unit test file.
+  -  Smaller tests are better.
+  -  Always use seed if you're using any stochastic functions.
+  -  Avoid using sleep in multi thread tests.
+  -  Avoid using external resource in tests e.g: S3, requests to external services: download model from repos etc.
+
+
+#### Ready to make a change ?
+This project is built using Python 3.9 and [Poetry](https://python-poetry.org/docs/#installation).
```

Comment:
In the `pyproject.toml`, the Python version is still `python = ">=3.8,<3.10"`

Reply:
I know but the build pipelines we have setup use python 3.9 so I know it works with that for sure. 
Reference 
https://github.com/h1st-ai/h1st/blob/main/.github/workflows/publish.yml#L16
https://github.com/h1st-ai/h1st/blob/main/.github/workflows/test.yml#L29